src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/CloudTest.targets

```diff
@@ -96,25 +97,66 @@
     <Warning Condition="'$(EventHubPath)' != '' or '$(EventHubSharedAccessKey)' != '' or '$(EventHubSharedAccessKeyName)' != ''" Text="EventHubPath and EventHubAccessKeys are ignored, use HelixApiAccessKey instead" />
     <Error Condition="'$(SkipNotifyEvent)' != 'true' and '$(HelixApiAccessKey)' == ''" Text="HelixApiAccessKey must be set to start Helix jobs" />
 
-    <!-- gather the test archives for upload and filter based on FilterToTestTFM -->
+    <!-- gather the test archives for this build -->
     <ItemGroup>
-      <UnfilteredTestArchives Include="$(TestArchivesRoot)**/*.zip" >
+      <OSPlatformTestArchives Include="$(TestArchivesRoot)**/*.zip" >
         <OSPlatformConfig>$(OSPlatformConfig)</OSPlatformConfig>
-      </UnfilteredTestArchives>
-      <UnfilteredTestArchives Include="$(AnyOSTestArchivesRoot)**/*.zip" >
+      </OSPlatformTestArchives>
+      <AnyOSPlatformTestArchives Include="$(AnyOSTestArchivesRoot)**/*.zip" >
         <OSPlatformConfig>$(AnyOSPlatformConfig)</OSPlatformConfig>
-      </UnfilteredTestArchives>
+      </AnyOSPlatformTestArchives>
       <!-- Only include Unix folders if supported by the current target OS -->
-      <UnfilteredTestArchives Condition="'$(TargetsUnix)' == 'true'" Include="$(UnixTestArchivesRoot)**/*.zip" >
+      <UnixPlatformTestArchives Condition="'$(TargetsUnix)' == 'true'" Include="$(UnixTestArchivesRoot)**/*.zip" >
         <OSPlatformConfig>$(UnixPlatformConfig)</OSPlatformConfig>
-      </UnfilteredTestArchives>
+      </UnixPlatformTestArchives>
+    </ItemGroup>
+
+    <PropertyGroup>
+        <OSPlatformTestArchivesIndexFile>$(TestArchivesRoot)/$(OSPlatformConfig)-Test-Archive-Index.txt</OSPlatformTestArchivesIndexFile>
+        <AnyOSPlatformTestArchivesIndexFile>$(AnyOSTestArchivesRoot)/$(AnyOSPlatformConfig)-Test-Archive-Index.txt</AnyOSPlatformTestArchivesIndexFile>
+        <UnixPlatformTestArchivesIndexFile>$(UnixTestArchivesRoot)/$(UnixPlatformConfig)-Test-Archive-Index.txt</UnixPlatformTestArchivesIndexFile>
+    </PropertyGroup>
+
+    <!-- This is where we craft our archive indexes. These will sit at the base of the uploaded package. -->
```

Comment:
This sentence is confusing.  Maybe "in the root of the storage container for uploaded Zip files"